resources/emergency_contacts.py

```diff
@@ -1,89 +1,67 @@
 
     def get(self, id=None):
-        # GET /emergencynumbers
-        if not id:
-            return {'emergency_contacts': [e.json() for e in EmergencyContactModel.query.all()]}
-
-        # GET /emergencynumbers/<id>
-        emergencyEntry = EmergencyContactModel.find_by_id(id)
-        if not emergencyEntry:
-            return {'message': 'Emergency contact not found'}, 404
-        return emergencyEntry.json()
+        if id:
+            return EmergencyContactModel.find(id).json()
+        else:
+            return {'emergency_contacts': [e.json() for e in EmergencyContactModel.query.all()]}, 200
 
     @admin_required
     def post(self):
-        data = EmergencyContacts.parser.parse_args()
-        if EmergencyContactModel.find_by_name(data["name"]):
-            return {'message': 'An emergency contact with this name already exists'}, 401
-
-        # In the JSON body, contact_numbers is expected to be an array of dictionaries
-        # But, reqparser is not able to extract nested JSON data (see https://github.com/flask-restful/flask-restful/issues/517)
-        numbersData, numbersError = parseContactNumbersFromJson(request.get_json(force=True))
-        if numbersError:
-            return {'message': numbersError}, 401
-        data["contact_numbers"] = numbersData
-
-        contactEntry = EmergencyContactModel(**data)
-        
-        EmergencyContactModel.save_to_db(contactEntry)
-        
-
-        return contactEntry.json(), 201
+        return EmergencyContactModel.create(EmergencyContactSchema, request.json).json(), 201
 
     @admin_required
     def put(self, id):
         parser_for_put = EmergencyContacts.parser.copy()
-        parser_for_put.replace_argument('name',required=False)
-        parser_for_put.replace_argument('contact_numbers',action='append',required=False)
+        parser_for_put.replace_argument('name', required=False)
+        parser_for_put.replace_argument('contact_numbers', action='append', required=False)
         data = parser_for_put.parse_args()
 
         contactEntry = EmergencyContactModel.find_by_id(id)
         if not contactEntry:
             return {'message': 'Emergency contact not found'}, 404
 
-        #variable statements allow for only updated fields to be transmitted
-        if(data.name):
+        # variable statements allow for only updated fields to be transmitted
+        if (data.name):
             contactEntry.name = data.name
-        if('description' in data.keys()):
+        if ('description' in data.keys()):
             contactEntry.description = data.description if data.description else ""
-        if(data.contact_numbers):
-            numbersData, numbersError = parseContactNumbersFromJson(request.get_json(force=True))
+
+        # TODO: We should/need to create a ContactNumber resource to update this
+        if (data.contact_numbers):
+            numbersData, numbersError = parse_contact_numbers(request.get_json(force=True))
```

Comment:
Since we are not going to refactor the PUT action right now we should keep everything that it depends on. If we put the `parseContactNumbersFromJson` method back in and change this back to use that method then all the tests will pass and we can get this merged.
```suggestion
            numbersData, numbersError = parseContactNumbersFromJson(request.get_json(force=True))
```